File: Programmers/연습문제/Level 1/문자열 내 마음대로 정렬하기/Solution.js
function solution(strings, n) {
    const answer = getAnswer(strings, n);
    return answer;
}

function getAnswer(strings, n) {
    return strings.sort(
        (a, b) => a[n] === b[n] ? a.localeCompare(b) : a[n].localeCompare(b[n])
    )
}


// TC
const strings = ["abce", "abcd", "cdx"];
const n = 2;
console.log(solution(strings, n));

File: Programmers/연습문제/Level 1/문자열 내 마음대로 정렬하기/Solution.swift
func solution(_ strings:[String], _ n:Int) -> [String] {
    return strings.sorted{
        Array($0)[n] != Array($1)[n] ? Array($0)[n] < Array($1)[n] : $0 < $1
    }
}


// TC
let strings = ["abce", "abcd", "cdx"]
let n = 2
print(solution(strings, n))
